docs/runtime-architecture.md

```diff
@@ -0,0 +1,57 @@
+# Runtime Architecture
+## Generated Libraries
+
+When using `jsii-pacmak` to generate libraries in different programming
+languages, the **Javascript** code is bundled within the generated library, so
+that it can be used during at runtime. This is the reason why a `node` runtime
+needs to be available in order to execute code that depends on `jsii` libraries.
+
+The generated libraries have a dependency on a *Runtime* library for the
+language, which contains the necessary logic to start a sidekick `node` process,
+load the `jsii-kernel` and then interact with it.
+
+## Architecture Overview
+
+A simplified representation of the execution environment of an application using
+`jsii` libraries from a different language follows:
+
+```
+                           ┌──────────────────────┐
+                           │                      │
+                           │     User's Code      │
+                           │                      │
+┌──────────────────────┐   ├──────────────────────┤
+│                      │   │                      │
+│User's Javascript Code│   │  Generated Bindings  │
+│                      │   │                      │
+├──────────────────────┤   ├──────────────────────┤
+│                      │   │                      │
+│     jsii Kernel      ◀━━━▶   jsii Runtime Lib   │
+│                      │   │                      │
+├──────────────────────┤   ├──────────────────────┤
+│                      │   │                      │
+│         node         │   │   JVM / .NET / ...   │
```

Comment:
It's important to depict that the node process is a child of the host, and that there is a 1:1 relationship.